### ISSUE TITLE
Reduce_mean in train losses

### ISSUE THREAD

**Issue by antoinedemathelin** [adapt-python/adapt#23]
The use of ``tf.reduce_mean`` in train losses could lead to strange behavior.
We should try to keep loss by sample and let tensorflow make the aggregation.
